pkg/reconciler/revision/revision.go

```diff
@@ -96,6 +96,7 @@ func (c *Reconciler) reconcileDigest(ctx context.Context, rev *v1.Revision) erro
 		digestValue        string
 		containerName      string
 		isServingContainer bool
+		image              string
```

Comment:
I'd add the error to this struct and never return an error "properly". You can then use that error to set status.